packages/core/src/textures/resources/Resource.ts

```diff
@@ -194,17 +211,22 @@ export class Resource
      * @param {PIXI.GLTexture} glTexture - texture instance for this webgl context
      * @returns {boolean} `true` is success
      */
-    style(renderer, baseTexture, glTexture) // eslint-disable-line no-unused-vars
+    /* eslint-disable @typescript-eslint/no-unused-vars */
+    /* eslint-disable @typescript-eslint/ban-ts-ignore */
+    // @ts-ignore
+    style(renderer: Renderer, baseTexture: BaseTexture, glTexture: GLTexture): boolean
     {
         return false;
     }
+    /* eslint-enable @typescript-eslint/no-unused-vars */
+    /* eslint-enable @typescript-eslint/ban-ts-ignore */
 
     /**
      * Clean up anything, this happens when destroying is ready.
      *
      * @protected
      */
-    dispose()
+    dispose(): void
```

Comment:
This should be protected

Reply:
I don't know why it was protected, users should be able to call this thing,

Reply:
destroy is the public interface. Dispose is a single method resource can override to do any special cleanup. it should be protected

Reply:
My main concern was about ImageResource and it has ImageBitmap inside. `dispose()` is supposed to remove it, but compressed source is still there. We have `dispose()` methods not only for resources - for Geometries, Buffers and other things, and they are supposed to be recoverable.

Reply:
Okay public it is